client/grpcutil/grpcutil.go

```diff
@@ -90,11 +90,12 @@ func GetOrCreateGRPCConn(ctx context.Context, clientConns *sync.Map, addr string
 	if err != nil {
 		return nil, err
 	}
-	if old, ok := clientConns.Load(addr); ok {
+	if old, ok := clientConns.LoadOrStore(addr, cc); !ok {
+		// Successfully stored the connection.
+		return cc, nil
+	} else {
 		cc.Close()
 		log.Debug("use old connection", zap.String("target", cc.Target()), zap.String("state", cc.GetState().String()))
 		return old.(*grpc.ClientConn), nil
```

Comment:
```suggestion
	}
	cc.Close()
	log.Debug("use old connection", zap.String("target", cc.Target()), zap.String("state", cc.GetState().String()))
	return old.(*grpc.ClientConn), nil
```

Reply:
In that case, I need to move old, ok := clientConns.LoadOrStore(addr, cc) outside, otherwise old is invisible to outside
         old, ok := clientConns.LoadOrStore(addr, cc)
	if !ok {
		// Successfully stored the connection.
		return cc, nil
	}
       cc.Close()
      log.Debug("use old connection", zap.String("target", cc.Target()), zap.String("state", cc.GetState().String()))
      return old.(*grpc.ClientConn), nil

Reply:
changed.

Reply:
actually changed